python/ray/data/dataset.py

```diff
@@ -3628,6 +3630,45 @@ def _aggregate_result(self, result: Union[Tuple, TableRow]) -> U:
         else:
             return result
 
+    def _repr_html_(self) -> str:
+        try:
+            from tabulate import tabulate
+        except ImportError:
+            logger.info("For rich Dataset reprs in notebooks, `pip install tabulate`.")
+            return ""
```

Comment:
`tabulate` is a Tune dependency. If users run `pip install 'ray[air]'`, they won't run into this error. That being said, I think this snippet is fine given that `tabulate` isn't a Datasets dependency.

Reply:
Yep, and initially I had it just directly importing `tabulate` - but a few of the CI jobs run tests in an environment that doesn't have it installed, so I realized it probably shouldn't be a hard dependency.